File: frontend/src/api/explain.ts
import axios from 'axios'

const api = axios.create({
  baseURL: '/api',
})

export interface ExplainRequest {
  code: string
  file_path?: string
  language?: string
}

export interface ExplainResponse {
  explanation: string
  complexity?: string
  issues?: string[]
}

export async function explainCode(request: ExplainRequest): Promise<ExplainResponse> {
  const response = await api.post<ExplainResponse>('/explain', request)
  return response.data
}

File: backend/models/explain.py
"""
Explain models
"""
from pydantic import BaseModel
from typing import Optional


class ExplainRequest(BaseModel):
    code: str
    file_path: Optional[str] = None
    language: Optional[str] = None  # "python", "javascript", etc.
    project_id: Optional[int] = None  # Optional project ID for context-aware explanation


class ExplainResponse(BaseModel):
    explanation: str
    complexity: Optional[str] = None
    issues: Optional[List[str]] = []

